libs/ha/ha_common_libs_k8s.py

```diff
@@ -1701,3 +1701,33 @@ def poll_to_get_resource_status(self, exp_sts, rsc, rsc_id,
         if status != exp_sts:
             return False
         return True
+
+    def failover_pod(self, pod_obj, pod_yaml, failover_node):
+        """
+        Helper function to delete, recreate and failover pod
+        :param pod_obj: Object of the pod
+        :param pod_yaml: Dict containing pod name and yaml file to be used for recreation
+        :param failover_node: Node FQDN to which pod is to be failed over
+        :return: bool, tuple
+        """
+        for pod, yaml_f in pod_yaml.items():
+            LOGGER.info("Deleting deployment for pod %s", pod)
+            resp = pod_obj.delete_deployment(pod_name=pod)
+            if resp[0]:
+                return False, f"Failed to delete pod {pod} by deleting deployment"
+            deploy = resp[2]
+            LOGGER.info("Recreating deployment for pod %s using yaml file %s", pod, yaml_f)
+            resp = pod_obj.recover_deployment_k8s(yaml_f, deploy)
+            if not resp[0]:
+                return resp
+            new_pod = pod_obj.get_recent_pod_name(deployment_name=deploy)
+            LOGGER.info("Changing host node of pod %s to %s", pod, failover_node)
+            resp = self.change_pod_node(pod_obj, [new_pod], failover_node)
```

Comment:
Maybe on the change_pod_node method, if given a list of pods, then we would failover all of them to one single failover_node?

Reply:
Yes, change_pod_node is designed that way. If we give list of pods for which we want to change NodeSelector to particular node, this lib can be used.
For e.g. Change node for HA and control pod.

Reply:
Added.